libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationNotificationProvider.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import android.content.Context
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation
+import com.mapbox.services.android.navigation.v5.navigation.notification.NavigationNotification
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+
+internal class NavigationNotificationProvider(applicationContext: Context, mapboxNavigation: MapboxNavigation) {
+
+    private var navigationNotification: NavigationNotification? = null
```

Comment:
```suggestion
   var navigationNotification: NavigationNotification? = null
     private set 
```
and remove `fun retrieveNotification`

Reply:
👍